### PR TITLE
Improve helm chart: Allow to enable or disable cloudControllerManager/cloudNodeManager, configure ca certificates directory and replicas

### PR DESCRIPTION
**Pull request by forselli-stratio**
…/cloudNodeManager, configure ca certificates directory and replicas

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This PR allows to enable/disable both cloud-controller-manager or cloud-node-manager deployments. Also, you can configure a different directory for the CA certificates in cloud-controller-manager. Finally, it allows you to configure the number of replicas for the cloud-controller-manager.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: helm/cloud-provider-azure/Chart.yaml [hunk 1]
```diff
@@ -13,4 +13,4 @@ maintainers:
 name: cloud-provider-azure
 sources:
 - https://github.com/kubernetes-sigs/cloud-provider-azure
-version: 1.23.11
+version: 1.24.0
```
Comment (jackfrancis) [inline helm/cloud-provider-azure/Chart.yaml:16]: Let's skip the chart version update for now, we can queue up these changes and release them with the next batch of cloud-provider-azure releases (which will include 1.23.12)

File: helm/cloud-provider-azure/values.yaml [hunk 2]
```diff
@@ -30,6 +32,7 @@ cloudControllerManager:
   # minResyncPeriod: "100m"
   # nodeStatusUpdateFrequency: "10m"
   port: 10267
+  replicaCount: 1
```
Comment (jackfrancis) [inline helm/cloud-provider-azure/values.yaml:35]: slight preference to just using the value `replicas` here (to match the standard k8s spec)

**Comment by jackfrancis** [comment]
Thanks for this @forselli-stratio! Added some comments, overall looking good.

File: helm/cloud-provider-azure/values.yaml (hunk 2 above)
Comment (forselli-stratio) [reply helm/cloud-provider-azure/values.yaml]: sure, no problem

**Comment by forselli-stratio** [comment]
> Thanks for this @forselli-stratio! Added some comments, overall looking good.

Hi @jackfrancis , i updated the PR with the changes you sugested. Thank you!

**Comment by lzhecheng** [comment]
@forselli-stratio please fix the `pull-cloud-provider-azure-check` error. Thanks.

File: helm/cloud-provider-azure/README.md [hunk 3]
```diff
@@ -70,6 +72,7 @@ Below is the complete set of configuration that you may include when invoking `h
 | `cloudControllerManager.imagePullPolicy` | `"IfNotPresent"` | you may change to`"Always"` or `"Never"` if appropriate for your environment, see [here](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy) for more info |
 | `cloudControllerManager.logVerbosity` | `"2"` | set to a higher number when debugging the azure-cloud-controller-manager runtime |
 | `cloudControllerManager.port` | `"10267"` | TCP port on which azure-cloud-controller-manager pod responds to requests |
+| `cloudControllerManager.replicaCount` | `1` | Number of replicas for the azure-cloud-controller-manager deployment. |
```
Comment (lzhecheng) [inline helm/cloud-provider-azure/README.md:75]:
```suggestion
| `cloudControllerManager.replicaCount` | `1` | Number of replicas for the azure-cloud-controller-manager deployment. It should be no more than the number of control plane Nodes |
```
I suggest mentioning this.

Comment (lzhecheng) [reply helm/cloud-provider-azure/README.md:75]: Also, isn't it `cloudControllerManager.replicas`?

**Comment by lzhecheng** [comment]
> @forselli-stratio please fix the `pull-cloud-provider-azure-check` error. Thanks.

The failure results from verify helm script. Not this PR.

**Comment by jackfrancis** [comment]
@MartinForReal @lzhecheng @forselli-stratio 

FYI: the failure of the `pull-cloud-provider-azure-check` PR is in fact due to this PR. As documented here, every time you make an update to the helm chart, you must run the `hack/update-helm-repo.sh` script to publish the new chart release to the repo:

[hack/update-helm-repo.sh](https://github.com/kubernetes-sigs/cloud-provider-azure/tree/master/helm/cloud-provider-azure#maintaining-the-repo)

I also realize that my prior guidance around not cutting a new version with these changes is wrong, apologies for that.

I have a new PR that attempts to clarify documentation around helm chart maintenance here:

https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1658

**Comment by jackfrancis** [comment]
So, @forselli-stratio I was wrong about not cutting a new version. Can we update the version in `Chart.yaml` to "1.23.11+20220505" (assuming you make these changes today, May 5).

After updating the version, please run `make update-helm` to render the new chart updates as a release in the local repo.

Thanks!

File: helm/cloud-provider-azure/templates/cloud-provider-azure.yaml [hunk 4]
```diff
@@ -277,10 +278,12 @@ spec:
             path: /etc/kubernetes
         - name: etc-ssl
```
Comment (nilo19) [inline helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:279]: would you mind changing this name as well?

Comment (jackfrancis) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml:279]: Now that the path is configurable we could change the name key to "ssl-mount" (remember to change both instances)

Thanks!

File: helm/cloud-provider-azure/README.md (hunk 3 above)
Comment (forselli-stratio) [reply helm/cloud-provider-azure/README.md]: Fixed

File: helm/cloud-provider-azure/templates/cloud-provider-azure.yaml (hunk 4 above)
Comment (forselli-stratio) [reply helm/cloud-provider-azure/templates/cloud-provider-azure.yaml]: Done

**Comment by forselli-stratio** [comment]
Hi @jackfrancis @MartinForReal @lzhecheng , i applied your suggestions. I noticed that 3 days ago you changed the version to 1.24.0, so i'm not sure if the version i used is correct. Thank you!!

File: helm/cloud-provider-azure/Chart.yaml [hunk 5]
```diff
@@ -13,4 +13,4 @@ maintainers:
 name: cloud-provider-azure
 sources:
 - https://github.com/kubernetes-sigs/cloud-provider-azure
-version: 1.24.0
+version: 1.24.1+20220509
```
Comment (jackfrancis) [inline helm/cloud-provider-azure/Chart.yaml:16]: This should be 1.24.**0**+<datestamp> because the current, latest release of the chart is 1.24.0.

Comment (forselli-stratio) [reply helm/cloud-provider-azure/Chart.yaml]: Done @jackfrancis , thanks!

**Comment by jackfrancis** [comment]
@forselli-stratio the versions are still not right. To correct:

1)

```
$ # remove the 1.24.1 chart from the repo
$ rm helm/repo/cloud-provider-azure-1.24.1+20220509.tgz
```

2)

```
$ # update the Chart.yaml to 1.24.0+<current date>
$ cat helm/cloud-provider-azure/Chart.yaml | grep 'version:'
version: 1.24.0+20220519
```

3)

```
$ # update the repo with the new 1.24.0 version
$ hack/update-helm-repo.sh 
Successfully packaged chart and saved it to: helm/repo/new/cloud-provider-azure-1.24.0+20220519.tgz
```

4)

```
$ # manually remove the 1.24.1 entry from index.yaml, something like this:
$ git diff helm/repo/index.yaml 
diff --git a/helm/repo/index.yaml b/helm/repo/index.yaml
index 351ee5134..058cc1b31 100644
--- a/helm/repo/index.yaml
+++ b/helm/repo/index.yaml
@@ -2,10 +2,10 @@ apiVersion: v1
 entries:
   cloud-provider-azure:
   - apiVersion: v1
-    created: "2022-05-09T13:30:32.197485547+02:00"
+    created: "2022-05-19T09:31:26.411883-07:00"
     description: A Helm chart for installing kubernetes-sigs/cloud-provider-azure
       components
-    digest: 081cc30534a47c6d0d420462cc81375453b56d1ca1ae1d3f17a7e7d6cc7f6508
+    digest: 72a94ef39f105f869fb9f40163fc96a19d8edcb1788746b3dae3dd3687fcaf8d
     home: https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/cloud-provider-azure/README.md
     keywords:
     - cloud-provider-azure
@@ -20,29 +20,8 @@ entries:
     sources:
     - https://github.com/kubernetes-sigs/cloud-provider-azure
     urls:
-    - cloud-provider-azure-1.24.1+20220509.tgz
-    version: 1.24.1+20220509
-  - apiVersion: v1
-    created: "2022-05-06T10:10:12.501287-07:00"
-    description: A Helm chart for installing kubernetes-sigs/cloud-provider-azure
-      components
-    digest: c9aa0bf18ee1442f9ae57ae277a6a7b1c7b88f33836508a6a97fb593cd1e1750
-    home: https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/cloud-provider-azure/README.md
-    keywords:
-    - cloud-provider-azure
-    - kubernetes-sigs
-    - kubernetes
-    maintainers:
-    - email: jackfrancis@gmail.com
-      name: Jack Francis
-    - email: zhechengli1995@outlook.com
-      name: Zhecheng Li
-    name: cloud-provider-azure
-    sources:
-    - https://github.com/kubernetes-sigs/cloud-provider-azure
-    urls:
-    - cloud-provider-azure-1.24.0.tgz
-    version: 1.24.0
+    - cloud-provider-azure-1.24.0+20220519.tgz
+    version: 1.24.0+20220519
   - apiVersion: v1
     created: "2022-05-02T14:33:56.285027-07:00"
     description: A Helm chart for installing kubernetes-sigs/cloud-provider-azure
@@ -106,4 +85,4 @@ entries:
     urls:
     - cloud-provider-azure-1.23.8.tgz
     version: 1.23.8
-generated: "2022-05-09T13:30:32.196495952+02:00"
+generated: "2022-05-19T09:31:26.411046-07:00"
```

**Comment by forselli-stratio** [comment]
Hi @jackfrancis , thanks for the detailed explanation, i hope i understood the flow right this time.

File: helm/cloud-provider-azure/README.md [hunk 6]
```diff
@@ -70,6 +72,7 @@ Below is the complete set of configuration that you may include when invoking `h
 | `cloudControllerManager.imagePullPolicy` | `"IfNotPresent"` | you may change to`"Always"` or `"Never"` if appropriate for your environment, see [here](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy) for more info |
 | `cloudControllerManager.logVerbosity` | `"2"` | set to a higher number when debugging the azure-cloud-controller-manager runtime |
 | `cloudControllerManager.port` | `"10267"` | TCP port on which azure-cloud-controller-manager pod responds to requests |
+| `cloudControllerManager.replicas` | `1` | Number of replicas for the azure-cloud-controller-manager deployment. It should be no more than the number of control plane Nodes |
```
Comment (lzhecheng) [inline helm/cloud-provider-azure/README.md:75]:
```suggestion
| `cloudControllerManager.replicas` | `"1"` | Number of replicas for the azure-cloud-controller-manager deployment. It should be no more than the number of control plane Nodes |
```

**Review by lzhecheng — COMMENTED** [review]
@forselli-stratio Thank you for the PR! Please rebase again and I think this PR should be merged then.

**Comment by forselli-stratio** [comment]
Hi @lzhecheng , done!!

**Pull request merged**